### PR TITLE
ESC-386

### PR DESCRIPTION
**Pull request by sam-miele**

### CODE REVIEW

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala [hunk 1]
```diff
@@ -117,8 +117,13 @@ class BusinessEntityController @Inject() (
       journeyTraverseService.getPrevious[BusinessEntityJourney].flatMap { previous =>
         store.get[BusinessEntityJourney].flatMap {
           case Some(journey) =>
-            val form = journey.eori.value.fold(eoriForm)(eori => eoriForm.fill(FormValues(eori)))
-            Ok(eoriPage(form, previous)).toFuture
+            val isEligible = journey.addBusiness.value.nonEmpty
```
Comment (priyankadhiman3006) [inline app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala:120]: can we move this val in the journey?

**Review by carwynellis — APPROVED** [review]
I think this is ok but I reckon pushing the logic down into the journeys might be a cleaner way to go and help simplify the controllers. Pinged you on slack.

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala (hunk 1 above)
Comment (carwynellis) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala:120]: Yeah my thoughts exactly. I've gone some way in this direction with the undertaking journey. Ideally we can just ask the journey what's next. Cleans up the controllers and makes it easier to unit test the journey logic.

**Review by priyankadhiman3006 — APPROVED** [review]

**Review by priyankadhiman3006 — APPROVED** [review]

**Review by carwynellis — APPROVED** [review]
Cheers for doing the updates. LGTM 👍

**Pull request merged**